src/werkzeug/http.py

```diff
@@ -422,11 +422,12 @@ def parse_dict_header(value: str, cls: type[dict] | None = None) -> dict[str, st
         value = value.decode("latin1")
 
     for item in parse_list_header(value):
-        key, has_value, value = item.partition("=")
+        # Base64-encoded objects can be right-padded with `=` characters
+        key, has_value, value = item.rstrip("=").partition("=")
         key = key.strip()
 
         if not has_value:
-            result[key] = None
+            result[item.strip()] = None
             continue
```

Comment:
My original idea was to run the `str.partition` again without the `rstrip` to avoid mangling `value` in cases where it is a valid key/value pair, that happens to end in `=`.

I thought this was overly paranoid, but it's super-trivial to make it work like that if you think such caution is warranted.